recipes-core/dist-nilrt/files/ptest/test_safemode_provisioning.sh

```diff
@@ -0,0 +1,130 @@
+		print_error "Partition ${label} @ ${device} does not match required label ${correct_re_label}"
+		found_error=true
+	fi
+	if [[ ! "${uuid}" =~ $correct_re_uuid ]]; then
+		print_error "Partition ${label} @ ${device} does not match required type UUID ${correct_re_uuid}"
+		found_error=true
+	fi
+}
+
+
+found_error=false
+
+if uname --machine | grep -q arm; then
+	echo "INFO: detected an ARM machine architecture." \
+		  "This ptest is only valid for non-ARM, RAUC systems."
+	ptest_skip
+fi
+
+
+## PARTITION TESTS ##
+echo "## Checking that RAUC partitons are present..."
+# The UUIDs here should match those set by the disk_config_x64 file.
+# https://github.com/ni/meta-nilrt/blob/983d50796302b394aa428b992dae48b4da32ff08/recipes-core/initrdscripts/files/disk_config_x64#L28
+RE_ESP_UUID=C12A7328\-F81F\-11D2\-BA4B\-00A0C93EC93B
+RE_X64_OTHER_UUID=0FC63DAF\-8483\-4772\-8E79\-3D69D8477DE4
+
+# Determine the configured /boot storage device. Assert that this device should
+# be the root of the partitions we evaluate in the remainder of this test.
+boot_part=$(findmnt --kernel --first-only --noheadings --output SOURCE /boot)
+test -n "${boot_part}" || { print_error "No detected /boot mount; system state invalid."; ptest_fail; }
+boot_dev=/dev/$(lsblk -no PKNAME ${boot_part})
+test -e "${boot_dev}" || { print_error "Boot device ${boot_dev} does not exist; system state invalid."; ptest_fail; }
+
+
+readarray -t parts < <(fdisk -l ${boot_dev} -o Device,Name,Type-UUID | grep -o -E '^\/dev\/.*')
+
+# ASSERT: Only 4 partitions exist
+expected_partitions=4
+test ${#parts[@]} = $expected_partitions || { print_error "Expected $expected_partitions partitions, found ${#parts[@]}"; ptest_fail; }
+
+# ASSERT: part 0 is: nigrub and is an ESP partition
+test_part_id ${parts[0]} nigrub $RE_ESP_UUID
+# ASSERT: part 1 is: nibootfs and is a data partition type
+test_part_id ${parts[1]} nibootfs $RE_X64_OTHER_UUID
+# ASSERT: part 2 is: niconfig and is a data partition type
+test_part_id ${parts[2]} niconfig $RE_X64_OTHER_UUID
+# ASSERT: part 3 is: nirootfs and is a data partition type
+test_part_id ${parts[3]} nirootfs $RE_X64_OTHER_UUID
+
+part_1=( ${parts[1]} )
+test $boot_part = ${part_1[0]} || { print_error "booted from ${part_1[0]} partition instead of nibootfs"; ptest_fail; }
+
+# bail out if the partition tests failed
+$found_error && ptest_fail
+
+
+## PARTITION CONTENTS TESTING ##
+echo "## Checking that nibootfs contents are as expected..."
+safemode_files=(                       \
+	/boot/bootmode                      \
+	/boot/.safe/bzImage                 \
+	/boot/.safe/bootimage.ini           \
+	/boot/.safe/EFI_NI_vars             \
+	/boot/.safe/GRUB_NI_readonly_vars   \
+	/boot/.safe/bootimage.cfg           \
+	/boot/.safe/SMBIOS_NI_vars          \
+	/boot/.safe/ramdisk.gz              \
+	/boot/.safe/ramdisk.xz              \
+	/boot/grub/recoverytool-ni-version  \
+	/boot/grub/grub-ni-version          \
+	/boot/grub/grubenv                  \
```

Comment:
Style: I don't think it's worthwhile to try and align the line-escape characters like this.
1. It means that every time someone adds an entry which is even 1 character longer than the previous-longest entry, their diff will have to touch every entry to fixup the alignment.
2. The first line in this assignment block (`safemode_files=...`) is only ever going to align to the array elements coincidentally, because all the entry lines begin with a tab character - which is appropriate for bash styling.

Reply:
Good point. But it's easier to parse through the lines when the '\'s are not in the way. And we don't expect to add anymore lines to this list as safemode is frozen.
You're right about the first line but since we never change how many tab characters would go before the array element, and we don't expect to add new lines, so this wouldn't need to be realigned in future.